### ISSUE TITLE
Add Chronicle Queue support

### ISSUE THREAD

**Issue by gortiz** [akka/alpakka#258]
[Chronicle Queue](https://github.com/OpenHFT/Chronicle-Queue) is a micro second messaging queue that stores data offheap (in fact on disk by using mmap). At [ToroDB](https://github.com/torodb/) we use Akka Streams replicate data from MongoDB to a SQL database by using Akka Stream and we would like add Chronicle Queue as a Akka Stream offheap buffer/flow, but we think it would also be useful to have Sources/Sinks connected to a Chronicle Queue.

As we really want to do that and, as far as we know, there is no integration yet, we are definitely going to write a solution. We would like to contribute with it to Alpakka project, but as we are mainly Java developers used to work with Maven, I'm not sure if we are going to be able to fulfill all contributing guidelines. On the worst case scenario, we will release the code under BSD or Apache 2 license, so it should be possible to integrate it later.

**Comment by johanandren** [comment]
Java connectors are welcome, we could probably just ask someone Scala savvy to provide thin wrappers/factories for the Scala side of things.

If it still is a roadblock you could also develop and publish it as your own project and have it listed in the "External connectors" docs.

**Comment by gortiz** [comment]
After a couple hours of developmente, I reached a solution. I have to test it and verify the choosen API... but @akara, developer of Scubs, told me on twitter about their solution: https://github.com/paypal/squbs/blob/master/docs/persistent-buffer.md

It seems to be just what I was looking for. It may be more _heavy_ than I would like (I couldn't check its size and dependencies) and it looks like Scala 2.12 is not supported yet, but It could be interesting to add it as an _external connector_, right?

**Comment by 2m** [comment]
Yes. It would be great to have it listed there for better discoverability.

**Comment by akara** [comment]
Will take care of listing once we have it all released (should be a matter of days now). As for the weight, most of the dependencies in that module should be "provided" and intransitive so we do not deal with the weight of the dependencies. I'll check that.

**Comment by akara** [comment]
Also, yes, we're working on Scala 2.12 in the 1.0 release. But this can be months out.

**Comment by 2m** [comment]
Sounds good!

**Comment by gortiz** [comment]
As far as I know, Squbs does provide a flow that acts like a buffer but not sources or sinks. Therefore, I'm still working on custom stages to support that, as we are going to need them on ToroDB. I expect to have something to show you before the end of this week.

**Comment by akara** [comment]
I generally would not do a sink as it is no different from a `Flow` connected to a `Sink.ignore`. Doing a `Source` is a valid use case, though.

**Comment by gortiz** [comment]
Good point, but I think it may be different. It can be more efficient, as you don't need to recreate the object to just ignore it later. I guess these objects will be easily collected by the GC (as they should always be on the eden) but it is very easy to have a source once you implemented a flow.
